modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala

```diff
@@ -204,7 +236,18 @@ trait DataApplicationL0Service[F[_], D <: DataUpdate, DON <: DataOnChainState, D
 
 trait DataApplicationL1Service[F[_], D <: DataUpdate, DON <: DataOnChainState, DOF <: DataCalculatedState]
     extends DataApplicationService[F, D, DON, DOF]
-    with DataApplicationL1ContextualOps[F, D, DON, DOF]
+    with DataApplicationL1ContextualOps[F, D, DON, DOF] {
+
+  def requestDecoder(implicit a: Async[F]): EntityDecoder[F, NonEmptyList[Signed[DataTransaction]]] = {
+    implicit val decoderDataUpdate: Decoder[Signed[D]] = Signed.decoder[D](dataDecoder)
```

Comment:
either `decodeDataUpdate` or `dataUpdateDecoder` (for following decoders/encoders as well)

Reply:
👍🏼